src/test/shell/bazel/starlark_repository_test.sh

```diff
@@ -926,6 +926,49 @@ EOF
       || fail "Expected unrelated action to not be rerun"
 }
 
+function test_repo_env_inverse() {
+  # This test makes sure that a repository rule that has no dependencies on
+  # environment variables does _not_ get refetched when --repo_env changes.
+  setup_starlark_repository
+
+  cat > test.bzl <<'EOF'
+def _impl(ctx):
+  # Record a time stamp to verify that the rule is not rerun.
+  ctx.execute(["bash", "-c", "date +%s >> env.txt"])
+  ctx.file("BUILD", 'exports_files(["env.txt"])')
+
+repo = repository_rule(
+  implementation = _impl,
+)
+EOF
+  cat > BUILD <<'EOF'
+genrule(
+  name = "repoenv",
+  outs = ["repoenv.txt"],
+  srcs = ["@foo//:env.txt"],
+  cmd = "cp $< $@",
+)
+EOF
+  cat > .bazelrc <<EOF
+build:foo --repo_env=FOO=foo
+build:bar --repo_env=FOO=bar
+EOF
+
+  bazel build --config=foo //:repoenv
+  cp `bazel info bazel-genfiles 2>/dev/null`/repoenv.txt repoenv1.txt
```

Comment:
`bazel-genfiles` is deprecated, please just use `bazel-bin`. 
See https://github.com/bazelbuild/bazel/issues/6761

Reply:
Good point.

Done